src/MoBi.Presentation/Presenter/Main/StatusBarPresenter.cs

```diff
@@ -172,5 +204,19 @@ public void Handle(ReportCreationFinishedEvent eventToHandle)
          _numberOfReportsBeingCreated--;
          updateReportInfo();
       }
+
+      public void Handle(SimulationRunFinishedEvent eventToHandle)
+      {
+         _simulations[eventToHandle.Simulation.Id] = false;
+         if (activeSimulations == 0)
+         {
+            resetCountersAndHideBar();
+         }
+      }
+
+      public void Handle(SimulationRunStartedEvent eventToHandle)
+      {
+         _simulations[eventToHandle.Simulation.Id] = true;
```

Comment:
are these the rigth methods to use with a concurrent dictionary? I though we were supposed to use AddOrUpdate etc.. Otherwise you may be using it differently. This needs to be checked